src/vmm/src/arch/mod.rs

```diff
@@ -56,7 +57,23 @@ pub struct InitrdConfig {
 pub const GUEST_PAGE_SIZE: usize = 4096;
 
 /// Default page size for the host OS.
-pub const HOST_PAGE_SIZE: usize = 4096;
+pub static mut HOST_PAGE_SIZE: usize = 4096;
+
+/// Updates the HOST_PAGE_SIZE global variable to the output of
```

Comment:
Instead of modifying a global, could we make it a function that returns the pagesize? Or use `lazy_static!`.

Reply:
added a method to return the host page size.

Reply:
Seems like we never added this method. Why? If you don't want to do a syscall in the initializer for IoVecDeque, you can just use a `LazyLock` or some other variant of `OnceCell` here

Reply:
There is an `update_host_page_size` method just bellow.

Reply:
I think Pablo's point was to use a function that returns the host page size via a call to sysconf each time, eliminating the mutable static (at least that's how I understood it. If not, then it is very much my point though xP). If that's not acceptable for performance reasons, then please use one of the stdlib types that safely abstract this pattern away, such as `LazyLock`:

```
pub static HOST_PAGE_SIZE: LazyLock<usize> = LazyLock::new(|| {
    let r = unsafe { libc::sysconf(libc::_SC_PAGESIZE) };
    if r < 0 {
        warn!("Could not get host page size with sysconf, assuming default 4K host pages");
        4096
    } else {
        r as usize
    }
});
```

Reply:
LazyLock needs unstable `lazy_cell` feature. Updated to use `Once` instead.

Reply:
now with new Rust version `OnceLock` is available and I adjusted the implementation to use it.

Reply:
ah right. forgot that's what we were waiting for :rofl: thanks!